_episodes/05-exporting-data-2.md

```diff
@@ -0,0 +1,132 @@
+it's a
+good format to work with for maximum portability and endurance. Most spreadsheet programs can save to delimited text formats like CSV
+easily, although they may give you a warning during the file export.
+
+To save a file you have opened in Excel in CSV format:
+
+1. From the top menu select 'File' and 'Save as'.
+2. In the 'Format' field, from the list, select 'Comma Separated Values' (`*.csv`).
+3. Double check the file name and the location where you want to save it and hit 'Save'.
+
+An important note for backwards compatibility: you can open CSV files in Excel!
+
+![Saving an Excel file to CSV](../fig/excel-to-csv.png)
+
+## A Note on Cross-platform Operability
+
+By default, most coding and statistical environments expect UNIX-style line endings (`\n`) as representing line breaks.  However, Windows uses an alternate line ending signifier (`\r\n`) by default for legacy compatibility with Teletype-based systems.  
+
+As such, when exporting to CSV using Excel, your data in text format will look like this:
+
+>data1,data2\r\n1,2\r\n4,5\r\n
+
+When opening your CSV file in Excel again, it will parse it as follows:
+
+<img width="307" alt="screen shot 2017-03-31 at 7 15 07 pm" src="https://cloud.githubusercontent.com/assets/13110354/24560663/db4a5786-1643-11e7-931a-ca2c72336878.png">
+
+However, if you open your CSV file on a different system that does not parse the "\r" it will interpret your CSV file differently:
+
+Your data in text format then look like this:
+
+>data1 <br>
+>data2\r<br>
+>1<br>
+>2\r<br>
+>…
+
+This will then in turn parse as:
+
+<img width="308" alt="screen shot 2017-03-31 at 7 26 42 pm" src="https://cloud.githubusercontent.com/assets/13110354/24561066/a990327c-1645-11e7-90b5-35a44e90f8d9.png">
+
+thus causing terrible things to happen to your data.  For example, `2\r` is not a valid integer, and thus will throw an error (if you’re lucky) when you attempt to operate on it in R or Python.  Note that this happens on Excel for OSX as well as Windows, due to legacy Windows compatibility.
+
+There are a handful of solutions for enforcing uniform UNIX-style line endings on your exported CSV files:
+
+1. When exporting from Excel, save as a “Windows comma separated (.csv)” file
+2. If you store your data file under version control using Git, edit the `.git/config` file in your repository to automatically translate `\r\n` line endings into `\n`.
+Add the following to the file ([see the detailed tutorial](http://nicercode.github.io/blog/2013-04-30-excel-and-line-endings)):
+
+		[filter "cr"]
+    		clean = LC_CTYPE=C awk '{printf(\"%s\\n\", $0)}' | LC_CTYPE=C tr '\\r' '\\n'
+    		smudge = tr '\\n' '\\r'` 
+    
+ 	and then create a file `.gitattributes` that contains the line:
+ 
+ 		*.csv filter=cr
+ 
+	
+3. Use [dos2unix](http://dos2unix.sourceforge.net/) (available on OSX, *nix, and Cygwin) on local files to standardize line endings.
+
+#### A note on R and `xls`
```

Comment:
this is a little out of date. The `readxl` package didn't exist when this section was written and provides much greater compatibility than any of the preceding packages. We may want to encourage its use to create CSV files in a scripted way.